src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CommitManager.cs

```diff
@@ -69,6 +69,12 @@ internal sealed class CommitManager : IAsyncCompletionCommitManager
                 return CommitResultUnhandled;
             }
 
+            var filterText = session.ApplicableToSpan.GetText(subjectBuffer.CurrentSnapshot);
+            if (Controller.IsFilterCharacter(roslynItem, typeChar, filterText + typeChar))
+            { 
+                return new AsyncCompletionData.CommitResult(isHandled: true, AsyncCompletionData.CommitBehavior.CancelCommit);
+            }
```

Comment:
This is because the editor could not 100% properly decide if it is a commit behavior. We help them to decide with PotentialCommitCharacters / ShouldCommitCompletion but cannot guarantee. Here, in TryCommit we have more information to decide.

Originally, this decision was made in Controller.Type_Char.

---
In reply to: [226452525](https://github.com/dotnet/roslyn/pull/29016#discussion_r226452525) [](ancestors = 226452525)